### PR TITLE
SQL: Implement `CONVERT`, an alternative to `CAST`

### PR DESCRIPTION
**Pull request by matriv**
`CONVERT` works exactly like cast with slightly different syntax:
`CONVERT(<value>, <data_type)` as opposed to `CAST(<value> AS <data_type>)`

Moreover it support format of the MS-SQL data types `SQL_<type>`,
e.g.: `SQL_INTEGER`

Closes: #34513

### CODE REVIEW

**Comment by astefan** [comment]
Definitely UNSUPPORTED from usability point of view. It is not an invalid data type, but rather a valid one we do not support yet.

File: x-pack/plugin/sql/sql-proto/src/main/java/org/elasticsearch/xpack/sql/type/DataType.java [hunk 1]
```diff
@@ -44,12 +45,63 @@
     DATE(        JDBCType.TIMESTAMP, Timestamp.class, Long.BYTES,        24,                24);
     // @formatter:on
 
+    public static final String MSSQL_DATATYPE_PREFIX = "SQL_";
```
Comment (costin) [inline x-pack/plugin/sql/sql-proto/src/main/java/org/elasticsearch/xpack/sql/type/DataType.java]: It's not so much MSSQL as ODBC spec.

**Review by costin — APPROVED** [review]
LGTM.
+1 on using UNSUPPORTED instead of null.

File: x-pack/qa/sql/src/main/resources/docs.csv-spec [hunk 2]
```diff
@@ -1164,6 +1164,33 @@ SELECT YEAR(CAST('2018-05-19T11:23:45Z' AS TIMESTAMP)) AS year;
...
+SELECT CONVERT('123', SQL_INTEGER) AS int;
+
+      int
+---------------
+123
+// end::conversionStringToIntConvertMSSQLDataType
+;
+
+
+conversionStringToIntConvertESDataType
```
Comment (astefan) [inline x-pack/qa/sql/src/main/resources/docs.csv-spec]: I would put this example first and the odbc one second, in order of familiarity/usage?

**Review by astefan — APPROVED** [review]
LGTM. Left just one minor comment related to docs.

File: x-pack/plugin/sql/sql-proto/src/main/java/org/elasticsearch/xpack/sql/type/DataType.java (hunk 1 above)
Comment (matriv) [reply x-pack/plugin/sql/sql-proto/src/main/java/org/elasticsearch/xpack/sql/type/DataType.java]: Sure, replaced all MSSQL with ODBC.

**Comment by matriv** [comment]
retest this please

**Comment by matriv** [comment]
@elasticmachine run gradle build tests

**Comment by matriv** [comment]
retest this please

**Comment by matriv** [comment]
retest this please

**Comment by matriv** [comment]
retest this please

**Comment by matriv** [comment]
retest this please

**Comment by costin** [comment]
Try merging master again - hopefully it fixes the external failure.

**Pull request merged**